compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2682,8 +2681,13 @@ define_print_and_forward_display! {
 fn for_each_def(tcx: TyCtxt<'_>, mut collect_fn: impl for<'b> FnMut(&'b Ident, Namespace, DefId)) {
     // Iterate all local crate items no matter where they are defined.
     let hir = tcx.hir();
-    for item in hir.items() {
-        if item.ident.name.as_str().is_empty() || matches!(item.kind, ItemKind::Use(_, _)) {
+    for id in hir.items() {
+        if matches!(hir.def_kind(id.def_id), DefKind::Use) {
+            continue;
+        }
+
+        let item = hir.item(id);
+        if item.ident.name.as_str().is_empty() {
```

Comment:
We can avoid accessing the interner by comparing the `Symbol` directly.
```suggestion
        if item.ident.name == kw::Empty {
```